core/src/sumeragi/mod.rs

```diff
@@ -204,25 +204,28 @@ impl SumeragiHandle {
     }
 
     /// Deposit a sumeragi network message.
-    pub fn incoming_message(&self, msg: MessagePacket) {
-        if msg.message.is_none() {
-            if let Err(error) = self.control_message_sender.try_send(ControlFlowMessage {
-                view_change_proofs: msg.view_change_proofs,
-            }) {
-                self.metrics.dropped_messages.inc();
-                error!(
-                    ?error,
-                    "This peer is faulty. \
-                     Incoming control messages have to be dropped due to low processing speed."
-                );
+    pub fn incoming_message(&self, msg: Message) {
```

Comment:
IMO we should split this into 2 functions and remove `Message` enum. There is no need to group messages in the enum anymore